### ISSUE TITLE
Query option parser should not include the middle property in $expand if that property is selected in $select

### ISSUE THREAD

**Issue by xuzhg** [OData/odata.net#1574]
*Short summary (3-5 sentences) describing the issue.*

### Reproduce steps

1. For only have $expand, for example: `$expand=HomeAddress/Cities`, the SelectExpandClause is:
* AllSelected=True
* SelectItems: one ExpandedNavigationSelectItem

2. For have $select and $expand, for example: `$select=Name&$expand=HomeAddress/Cities`, the SelectExpandClause is:
* AllSelected=False
* SelectItems: one ExpandedNavigationSelectItem & Two PathSelectItem (Name & HomeAddress)

3. for `$select=HomeAddress/Street&$expand=HomeAddress/Cities`, the SelectExpandClause is:
* AllSelected=False
* SelectItems: one ExpandedNavigationSelectItem & Two PathSelectItem (HomeAddress & HomeAddress/Cites )

### Expected result

1, 2 are correct, for 3, it should only include one PathSelectItem

### Actual result

* 3, it includes two PathSelectItem.

**Comment by xuzhg** [comment]
$select=Name, HomeAddress&$expand=HomeAddress/City
==> 
One : ExpandedNavSelectItem
Two:  PathSelectItem

**Comment by xuzhg** [comment]
@xuzhg More issue:

`~/Customers(1)?$select=HomeLocation($select=ZipCode,Street)`

`ZipCode` is navigation property.

This will return all navigation properties' links in the payload in full meta:

```json
{ 
   "@odata.context":"http://localhost:11001/odata/$metadata#People(HomeLocation)/$entity",
   "@odata.type":"#Microsoft.Test.E2E.AspNet.OData.NavigationPropertyOnComplexType.Person",
   "@odata.id":"http://localhost:11001/odata/People(1)",
   "@odata.editLink":"http://localhost:11001/odata/People(1)",
   "HomeLocation":{ 
      "@odata.type":"#Microsoft.Test.E2E.AspNet.OData.NavigationPropertyOnComplexType.Address",
      "Street":"110th",
      "ZipCode@odata.associationLink":"http://localhost:11001/odata/People(1)/HomeLocation/ZipCode/$ref",
      "ZipCode@odata.navigationLink":"http://localhost:11001/odata/People(1)/HomeLocation/ZipCode",
      "DetailCodes@odata.associationLink":"http://localhost:11001/odata/People(1)/HomeLocation/DetailCodes/$ref",
      "DetailCodes@odata.navigationLink":"http://localhost:11001/odata/People(1)/HomeLocation/DetailCodes"
   }
}
```

It's not correct.

**Comment by Sreejithpin** [comment]
I was looking at #1574: Query option parser should not include the middle property in $expand if that property is selected in $select

I tried executing SelectAndExpandOnComplexIncludesBoth Unit test having select and expand text as
const string selectClauseText = "Address/City";
const string expandClauseText = "Address/Residents"

which as per defect should have 3 SeleteItems in Select and Expand clause (2 pathitems and 1 expandeditem)

But after parsing and extracting I get only 1 expandeditem and 1 path item . 
Am I missing something? Can you please throw some light on this

**Comment by xuzhg** [comment]
@Sreejithpin are all issues mentioned here resolved?

**Comment by Sreejithpin** [comment]
> @Sreejithpin are all issues mentioned here resolved?

@xuzhg Yes, context uri was rewritten as per out discussion and standards (ie for selects with / like HomeAddress/Street, it will be HomeAddress/Street and not HomeAddress(Street)  ).
